ui/app/src/utils/path.ts

```diff
@@ -123,28 +123,38 @@ export function getIndividualPaths(path: string, rootPath?: string): string[] {
 }
 
 export function getPasteItemFromPath(path: string, paths: string[]): PasteItem {
-  const sourcePath = withoutIndex(path);
-  let lookup = {
-    [sourcePath]: {
-      path,
-      children: []
-    }
+  // create PasteItem with base path
+  let pasteItem = {
+    path,
+    children: []
   };
-  paths.forEach((path) => {
-    lookup[path] = {
-      path: path,
+
+  paths.forEach((path) => addToPasteItem(pasteItem, path));
+  return pasteItem;
+}
+
+function addToPasteItem(pasteItem: PasteItem, path: string) {
+  const parentPath = getParentPath(path);
+
+  if (pasteItem.path === path) {
+    // path already added as pasteItem's root.
+    return;
```

Comment:
Let's either move this condition to one of the other two or wrap the other 2 `if` statements inside this one.